src/language_features/completion.rs

```diff
@@ -70,17 +69,14 @@ pub fn editor_completion(
                 .and_then(|compl| compl.resolve_provider)
                 .unwrap_or(false)
             {
-                format!(
-                    "set-option window lsp_completions_selected_item {}; \
-                        lsp-completion-item-resolve-request true; ",
-                    completion_item_index
-                )
+                "lsp-completion-item-resolve-request true".to_string()
             } else {
                 "".to_string()
             };
-            let on_select = format!(
-                "{}info -markup -style menu -- %§{}§",
-                &maybe_set_index,
+            let on_select = formatdoc!(
+                "lsp-completion-item-selected {completion_item_index}
```

Comment:
You now set `lsp_completions_selected_item` even if the server does not
support the `completionItem/resolve` request. This means we'll send the
`completionItem/resolve` request anyway and gopls will respond with an error.

Since we currently only use `lsp_completions_selected_item` for
resolve-requests, we should only set it we actually want to amek resolve-requests.
Maybe we can improve the naming.

Reply:
ah yes, i was experimenting with using a LspCompletionSelected={index} hook, but decided against it. I do however think that it is useful (or at least more intuitive) to have this option always set, so i would rather do the lsp-completion-item-resolve-request another way.

Reply:
Yeah I see that the current implementation is surprising. 
I think a truly elegant solution would avoid even the shell call in `lsp-completion-item-resolve-request` if the server doesn't support `completionItem/resolve`.
This could be implemented by removing the `InsertCompletionHide` hook (that would call `lsp-completion-item-resolve-request`) if the server lacks support.

Until we have something like that, we can keep the current implementation or make it more intuititve by adding a boolean variable for the capability.